### PR TITLE
ImageProcessing: expose some functions and allows in-memory processing

### PR DESCRIPTION
**Pull request by CyaNinix**
Currently, most of image processing is done inside the ImageProcessingAtom gem, and only a few functions are exposed.

It's interesting to expose some other functions to be able to do some image processing in memory, for example for a texture array  gem which only write one file for multiple images.

### CODE REVIEW

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Include/Atom/ImageProcessing/ImageProcessingBus.h
Comment (gadams3) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Include/Atom/ImageProcessing/ImageProcessingBus.h:73]: Please add comments to this function

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Include/Atom/ImageProcessing/ImageProcessingBus.h
Comment (gadams3) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Include/Atom/ImageProcessing/ImageProcessingBus.h:91]: Please comment these functions, inputs and outputs

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.cpp
Comment (gadams3) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.cpp:626]: Please add braces to all loops and conditionals

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.h
Comment (gadams3) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.h:115]: Please add 'override' to all of these overridden virtual function definitions.

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp
Comment (gadams3) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp:236]: Please add braces

**Review by gadams3 — CHANGES_REQUESTED** [review]

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.h
Comment (CyaNinix) [reply Gems/Atom/Asset/ImageProcessingAtom/Code/Source/BuilderSettings/BuilderSettingManager.h:115]: Those are not virtual functions.

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp
Comment (santorac) [inline Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp:231]: Can you factor this out to share code with ConvertImageObject?

**Review by santorac — APPROVED** [review]
These changes look good to me. There is one place where I would like to see the code factored out, but it's not significant enough to block the PR so you have my approval either way.

File: Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp
Comment (CyaNinix) [reply Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.cpp:231]: I can, but this requires me to either forward declare ImageConvertProcess in ImageBuilderComponent.h (to declare a private function using it), or make an anonymous function inside ImageBuilderComponent.cpp.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**
